examples/differential_rotation/reprojected_map.py

```diff
@@ -0,0 +1,96 @@
+"""
+==================
+Reprojecting a Map
+==================
+
+How to apply differential rotation to a Map.
+
+The example uses the `~sunpy.coordinates.metaframes.RotatedSunFrame` coordinate
+metaframe in `sunpy.coordinates` to apply differential rotation to a
+Map.  See :ref:`sunpy-coordinates-rotatedsunframe` for more details on
+using `~sunpy.coordinates.metaframes.RotatedSunFrame`.
+
+.. note::
+   This example requires `reproject` 0.6 or later to be installed.
+
+"""
+from astropy.coordinates import SkyCoord
+import astropy.units as u
+from astropy.wcs import WCS
+import matplotlib.pyplot as plt
+import numpy as np
+from reproject import reproject_interp
+
+from sunpy.coordinates import (Helioprojective, RotatedSunFrame,
+                               transform_with_sun_center)
+from sunpy.data.sample import AIA_171_IMAGE
+from sunpy.map import Map, make_fitswcs_header
+
+##############################################################################
+# First, load an AIA observation.
+
+m = Map(AIA_171_IMAGE)
+in_time = m.date
+
+##############################################################################
+# Let's define the output frame to be five days in the future for an observer
+# at Earth (i.e., about five degrees offset in heliographic longitude compared
+# to the location of AIA in the original observation).
+
+out_time = in_time + 5*u.day
+out_frame = Helioprojective(observer='earth', obstime=out_time)
+
+##############################################################################
+# For the reprojection, the definition of the target frame can be
+# counter-intuitive.  We will be transforming from the original frame to the
+# `~sunpy.coordinates.metaframes.RotatedSunFrame` version of the output frame
+# with ``rotated_time`` set to the time of the original frame.
+
+rot_frame = RotatedSunFrame(base=out_frame, rotated_time=in_time)
+print(rot_frame)
+
+##############################################################################
+# Construct a WCS object for the output map with the target
+# ``RotatedSunHelioprojective`` frame specified instead of the regular
+# `~sunpy.coordinates.frames.Helioprojective` frame.
+# If one has an actual ``Map`` object at the desired output
+# time (e.g., the actual AIA observation at the output time), one can use the
+# WCS object from that ``Map`` object instead of constructing a custom WCS.
```

Comment:
Quick note as to how to do this?

```suggestion
# WCS object from that ``Map`` object (e.g. ``mymap.wcs``) instead of constructing a custom WCS.
```

Reply:
Done